lib/internal/modules/package_json_reader.js

```diff
@@ -51,19 +60,23 @@ function deserializePackageJSON(path, contents) {
     exists: true,
     pjsonPath,
     name,
-    main,
-    type,
-    // This getters are used to lazily parse the imports and exports fields.
-    get imports() {
-      const value = requiresJSONParse(plainImports) ? JSONParse(plainImports) : plainImports;
-      ObjectDefineProperty(this, 'imports', { __proto__: null, value });
-      return this.imports;
-    },
-    get exports() {
-      const value = requiresJSONParse(plainExports) ? JSONParse(plainExports) : plainExports;
-      ObjectDefineProperty(this, 'exports', { __proto__: null, value });
-      return this.exports;
-    },
+    ...(main != null && { main }),
+    ...(type != null && { type }),
```

Comment:
Because as it was before, empty fields are forced to be present but with no value:

```json
{
  "name": "foo",
  "exports": "./index.js"
}
```

```js
{
  name: 'foo'
  main: undefined,
  exports: './index.js',
}
```

The only result is hassle and gotchas: ex `'main' in pkg` → `true`, which is a lie ("main" is _not_ in the package.json).

And in tests, it forces you to add extra properties that don't matter:

```js
assert.deepStrictEqual(pkg, {
  name: 'foo'
  exports: './index.js',
});
```

Pedantically fails because `main: undefined` is not explicitly specified.